braindecode/augmentation/transforms.py

```diff
@@ -0,0 +1,1000 @@
+    spherical_splines : bool, optional
+        Whether to use spherical splines for the interpolation or not. When
+        `False`, standard scipy.interpolate.Rbf (with quadratic kernel) will be
+        used (as in the original paper). Defaults to True.
+    random_state: int | numpy.random.Generator, optional
+        Seed to be used to instantiate numpy random number generator instance.
+        Defaults to None.
+
+    References
+    ----------
+    .. [1] Krell, M. M., & Kim, S. K. (2017). Rotational data augmentation for
+       electroencephalographic data. In 2017 39th Annual International
+       Conference of the IEEE Engineering in Medicine and Biology Society
+       (EMBC) (pp. 471-474).
+    """
+
+    def __init__(
+        self,
+        probability,
+        sensors_positions_matrix,
+        magnitude=None,
+        mag_range=(0, 30),
+        axis='z',
+        max_degrees=15,
+        spherical_splines=True,
+        random_state=None
+    ):
+        if isinstance(sensors_positions_matrix, (np.ndarray, list)):
+            sensors_positions_matrix = torch.as_tensor(
+                sensors_positions_matrix
+            )
+        assert isinstance(sensors_positions_matrix, torch.Tensor),\
+            "sensors_positions should be an Tensor"
+        assert isinstance(max_degrees, Real) and max_degrees >= 0,\
+            "max_degrees should be non-negative float."
+        assert isinstance(axis, str) and axis in ['x', 'y', 'z'],\
+            "axis can be either x, y or z."
+        assert sensors_positions_matrix.shape[0] == 3,\
+            "sensors_positions_matrix shape should be 3 x n_channels."
+        assert isinstance(spherical_splines, bool),\
+            "spherical_splines should be a boolean"
+        self.sensors_positions_matrix = sensors_positions_matrix
+        self.axis = axis
+        self.spherical_splines = spherical_splines
+
+        # override max_degrees value when a magnitude is passed
+        if magnitude is not None:
+            min_val, max_val = mag_range
+            self.max_degrees = magnitude * max_val + (1 - magnitude) * min_val
+        else:
+            self.max_degrees = max_degrees
+
+        # Handle case of 0 degrees
+        if self.max_degrees == 0:
+            operation = identity
+        else:
+            operation = random_rotation
+
+        super().__init__(
+            probability=probability,
+            operation=operation,
+            magnitude=magnitude,
+            mag_range=mag_range,
+            axis=self.axis,
+            max_degrees=self.max_degrees,
+            sensors_positions_matrix=self.sensors_positions_matrix,
+            spherical_splines=self.spherical_splines,
+            random_state=random_state
+        )
+
+
+class RandomZRotation(RandomSensorsRotation):
+    """Interpolates EEG signals over sensors rotated around the Z axis
+    with an angle sampled uniformly between 0 and `max_degree`.
```

Comment:
Inconsistent with `max_degree` doc string.
```suggestion
    with an angle sampled uniformly between ``-max_degree`` and ``max_degree``.
```